core/domain/stats_services.py

```diff
@@ -317,165 +245,67 @@ def get_exploration_stats(exploration_id, exploration_version):
     state_hit_counts = exp_stats['state_hit_counts']
 
     return {
-        'improvements': get_state_improvements(
-            exploration_id, exploration_version),
         'last_updated': last_updated,
         'num_completions': exp_stats['complete_exploration_count'],
         'num_starts': exp_stats['start_exploration_count'],
         'state_stats': {
             state_name: {
                 'name': state_name,
-                'firstEntryCount': (
+                'first_entry_count': (
                     state_hit_counts[state_name]['first_entry_count']
                     if state_name in state_hit_counts else 0),
-                'totalEntryCount': (
+                'total_entry_count': (
                     state_hit_counts[state_name]['total_entry_count']
                     if state_name in state_hit_counts else 0),
+                'no_submitted_answer_count': (
+                    state_hit_counts[state_name].get('no_answer_count', 0)
+                    if state_name in state_hit_counts else 0),
+                'num_default_answers': count_top_state_rule_answers(
+                    exploration_id, state_name,
+                    exp_domain.DEFAULT_OUTCOME_CLASSIFICATION),
             } for state_name in exploration.states
         },
     }
 
 
-def record_answer(
-        exploration_id, exploration_version, state_name, rule_str, session_id,
-        time_spent_in_sec, params, answer_value):
+def record_answer(exploration, state_name, submitted_answer):
```

Comment:
Originally I think the idea here was to ensure some consistency with the interaction ID and state being passed in rather than trusting the caller to provide correct values, but it's a weak argument. Done.